src/cascadia/TerminalConnection/CTerminalHandoff.cpp

```diff
@@ -101,14 +106,16 @@ HRESULT CTerminalHandoff::EstablishPtyHandoff(HANDLE in, HANDLE out, HANDLE sign
 {
     try
     {
-        // Stash a local copy of _pfnHandoff before we stop listening.
+        std::unique_lock lock{ _mtx };
+
+        // s_StopListeningLocked sets _pfnHandoff to nullptr.
+        // localPfnHandoff is tested for nullness below.
+#pragma warning(suppress : 26429) // Symbol '...' is never tested for nullness, it can be marked as not_null (f.23).
         auto localPfnHandoff = _pfnHandoff;
 
         // Because we are REGCLS_SINGLEUSE... we need to `CoRevokeClassObject` after we handle this ONE call.
         // COM does not automatically clean that up for us. We must do it.
-        s_StopListening();
-
-        std::unique_lock lock{ _mtx };
+        LOG_IF_FAILED(s_StopListeningLocked());

```

Comment:
FYI this change is correct (i.e. holding the lock while calling `s_StopListening`).
Previously, `s_StopListening` would acquire the lock itself, so nothing has effectively changed here.